### ISSUE TITLE
Integrate tweets indexing with rest of Anserini

### ISSUE THREAD

**Issue by lintool** [castorini/Anserini#214]
Tweets indexing is currently a completely separate code path:
https://github.com/castorini/Anserini/blob/master/src/main/java/io/anserini/index/IndexTweets.java

And it's single threaded...

In fact, tweets use a completely different API for reading the collection:
https://github.com/castorini/Anserini/tree/master/src/main/java/io/anserini/document/twitter

We should better integrate tweet indexing with rest of Anserini

**Comment by Peilin-Yang** [comment]
In `IndexTweets.java`, we have many fields other than `id` and `contents` which we only support for document.
I just wanted to confirm that do we need to index these fields? 
I do not think adding them will be hard, but we need to augment the search functionality in order to match the document structure in the index.

**Comment by lintool** [comment]
We can subclass this: https://github.com/castorini/Anserini/blob/master/src/main/java/io/anserini/index/generator/LuceneDocumentGenerator.java

Right? So it'll be another `DocumentGenerator`?

But yes, it'll require refactoring the searcher a bit to make everything line up....

**Comment by Peilin-Yang** [comment]
There are some utils at https://github.com/castorini/Anserini/tree/master/src/main/java/io/anserini/document/twitter and these utils were heavily used by https://github.com/castorini/Anserini/tree/master/src/main/java/io/anserini/nrts 
The utils can be modified and merged into current ingestion workflow. But that mean we also need to change `nrts` accordingly.
Thoughts?

**Comment by Peilin-Yang** [comment]
The API search(query, filter, numHits) is no longer available. So in order to add a filter (which is required by tweet search for cutting tweets outside the time window) we have to use
```
BooleanQuery.Builder builder = new BooleanQuery.Builder();
builder.add(filter, BooleanClause.Occur.FILTER);
builder.add(query, BooleanClause.Occur.MUST);
query = builder.build();
```

But for reranking (e.g. using RM3) we need to convert the query back to something like `
query = useQueryParser ? queryParser.parse(queryString) : AnalyzerUtils.buildBagOfWordsQuery(FIELD_BODY, analyzer, queryString);`

**Comment by lintool** [comment]
Fixed as part of #215, which has been merged into master.
